api/events/enroll.py

```diff
@@ -62,21 +73,35 @@ async def enroll(
     Used by workers to get a new job to process. If there is no job
     available, request returns 204 (no content).
 
+    Returns 503 (service unavailable) if the database pool is almost full.
+    Processing jobs should never block user requests.
+
     Non-error response is returned because having nothing to do is not an error
     and we don't want to spam the logs.
     """
 
+    if not current_user.is_service:
+        raise ForbiddenException("Only services can enroll for jobs")
+
+    # source_topic
+    source_topic: str | list[str]
+
+    if isinstance(payload.source_topic, str):
+        source_topic = validate_source_topic(payload.source_topic)
+    else:
+        source_topic = [validate_source_topic(t) for t in payload.source_topic]
```

Comment:
You could make your life easier with this change:
```suggestion
    if isinstance(payload.source_topic, str):
        source_topic = [validate_source_topic(payload.source_topic)]
    else:
        source_topic = [validate_source_topic(t) for t in payload.source_topic]
```

Reply:
But I am saving several nanoseconds later when the topic is either evaluated as `topic = $1` in the case of string or `topic = ANY($1)` for lists in the SQL query :)